asapdiscovery-docking/asapdiscovery/docking/openeye.py

```diff
@@ -0,0 +1,375 @@
+                if (
+                    retcode == oedocking.OEDockingReturnCode_NoValidNonClashPoses
+                    and self.posit_inputs.allow_low_posit_prob
+                ):
+                    opts.SetPoseRelaxMode(oedocking.OEPoseRelaxMode_ALL)
+                    opts.SetMinProbability(self.posit_inputs.low_posit_prob_thresh)
+                    pose_res, retcode = self.run_oe_posit_docking(
+                        opts, pose_res, du, lig_oemol, self.posit_inputs.num_poses
+                    )
+
+                # try again allowing clashes
+                if (
+                    self.posit_inputs.allow_final_clash
+                    and retcode == oedocking.OEDockingReturnCode_NoValidNonClashPoses
+                ):
+                    opts.SetPoseRelaxMode(oedocking.OEPoseRelaxMode_ALL)
+                    opts.SetAllowedClashType(oedocking.OEAllowedClashType_ANY)
+                    pose_res, retcode = self.run_oe_posit_docking(
+                        opts, pose_res, du, lig_oemol, self.posit_inputs.num_poses
+                    )
+
+            if retcode == oedocking.OEDockingReturnCode_Success:
+                for result in pose_res.GetSinglePoseResults():
+                    posed_mol = result.GetPose()
+                    prob = result.GetProbability()
+
+                    posed_ligand = Ligand.from_oemol(posed_mol, **pair.ligand.dict())
+                    # set SD tags
+                    sd_data = {
+                        DockingResultCols.DOCKING_CONFIDENCE_POSIT.value: prob,
+                        DockingResultCols.POSIT_METHOD.value: POSIT_METHOD.reverse_lookup(
+                            self.posit_inputs.posit_method.value
+                        ),
+                    }
+                    posed_ligand.set_SD_data(sd_data)
+
+                    docking_result = POSITDockingResults(
+                        input_pair=pair,
+                        posed_ligand=posed_ligand,
+                        probability=prob,
+                        provenance=self.provenance(),
+                    )
+                    docking_results.append(docking_result)
+
+            else:
+                if error == "skip":
+                    print(
+                        f"docking failed for input pair with compound name: {pair.ligand.compound_name}, smiles: {pair.ligand.smiles} and target name: {pair.complex.target.target_name}"
+                    )
+                    docking_results.append(None)
+                elif error == "raise":
+                    raise ValueError(
+                        f"docking failed for input pair with compound name: {pair.ligand.compound_name}, smiles: {pair.ligand.smiles} and target name: {pair.complex.target.target_name}"
+                    )
+                else:
+                    raise ValueError(f"Unknown error handling option {error}")
+
+        return docking_results
+
+    @staticmethod
+    def write_docking_files(
```

Comment:
Should probably be a method of the base class, thats my original omission.

Reply:
done here: https://github.com/choderalab/asapdiscovery/pull/594/commits/e38cd3cc7db87b5fbedfad964adfd39e58435464